src/solvers/smt/smt_casts.cpp

```diff
@@ -572,13 +572,22 @@ smt_astt smt_convt::convert_typecast_to_struct(const typecast2t &cast)
 smt_astt smt_convt::convert_typecast(const expr2tc &expr)
 {
   const typecast2t &cast = to_typecast2t(expr);
+
   if(cast.type == cast.from->type)
     return convert_ast(cast.from);
 
   // Casts to and from pointers need to be addressed all as one
   if(is_pointer_type(cast.type))
     return convert_typecast_to_ptr(cast);
 
+  // FAM Initialization?
```

Comment:
> Why the add2t here?

The issue is that initializing an fam directly with the "{}" creates an typecast that ESBMC cannot handle properly. (becoming an typecast.from "add" to "array"). I've done this to prevent having side-effects on other kinds of structures.

> Are we sure that the flexible array is initialized to zero?

No, but it should be an 0-sized array anyway. After this PR I will file an issue to properly handle 0-sized arrays, which should fix this.